From 18b5071694060bb2edd09a131273d140e54164be Mon Sep 17 00:00:00 2001
From: Hu Xile <huxile@oppo.com>
Date: Tue, 2 Dec 2025 17:20:07 +0800
Subject: [PATCH 13/15] ANDROID: Update symbols to oplus symbol list.

Add symbol list for oplus in gki/aarch64/symbols/oplus.

1 function symbol(s) added
  'int __traceiter_android_vh_scx_restore_flags(void *,
   const struct sched_class *, const struct sched_class *, int *)'

1 variable symbol(s) added
  'struct tracepoint __tracepoint_android_vh_scx_restore_flags'

Bug: 465306289
Bug: 470903229
Change-Id: I252c6880c022b61deb5a41d19c778c3376353752
Signed-off-by: Hu Xile <huxile@oppo.com>
(cherry picked from commit 235abe2e02b8f0af5687c08c067724591801f3ff)
---
 gki/aarch64/abi.stg       | 28 ++++++++++++++++++++++++++++
 gki/aarch64/symbols/oplus |  2 ++
 2 files changed, 30 insertions(+)

diff --git a/gki/aarch64/abi.stg b/gki/aarch64/abi.stg
index 04cb46d47ffa..e89d3e5bd7ae 100644
--- a/gki/aarch64/abi.stg
+++ b/gki/aarch64/abi.stg
@@ -411813,6 +411813,14 @@ function {
   parameter_id: 0x18bd6530
   parameter_id: 0x334927f6
 }
+function {
+  id: 0x9b635ad2
+  return_type_id: 0x6720d32f
+  parameter_id: 0x18bd6530
+  parameter_id: 0x3029eaf7
+  parameter_id: 0x3029eaf7
+  parameter_id: 0x13580d6c
+}
 function {
   id: 0x9b639aa2
   return_type_id: 0x6720d32f
@@ -438825,6 +438833,15 @@ elf_symbol {
   type_id: 0x9a33392f
   full_name: "__traceiter_android_vh_scx_ops_enable_state"
 }
+elf_symbol {
+  id: 0xb625db04
+  name: "__traceiter_android_vh_scx_restore_flags"
+  is_defined: true
+  symbol_type: FUNCTION
+  crc: 0x0b13dee1
+  type_id: 0x9b635ad2
+  full_name: "__traceiter_android_vh_scx_restore_flags"
+}
 elf_symbol {
   id: 0x26dcb47e
   name: "__traceiter_android_vh_scx_set_cpus_allowed"
@@ -446070,6 +446087,15 @@ elf_symbol {
   type_id: 0x18ccbd2c
   full_name: "__tracepoint_android_vh_scx_ops_enable_state"
 }
+elf_symbol {
+  id: 0xad5c16f2
+  name: "__tracepoint_android_vh_scx_restore_flags"
+  is_defined: true
+  symbol_type: OBJECT
+  crc: 0x7c5aa8a7
+  type_id: 0x18ccbd2c
+  full_name: "__tracepoint_android_vh_scx_restore_flags"
+}
 elf_symbol {
   id: 0xfbb545c8
   name: "__tracepoint_android_vh_scx_set_cpus_allowed"
@@ -518502,6 +518528,7 @@ interface {
   symbol_id: 0x39b3e607
   symbol_id: 0x59bde75a
   symbol_id: 0xf69f852d
+  symbol_id: 0xb625db04
   symbol_id: 0x26dcb47e
   symbol_id: 0xc4e90767
   symbol_id: 0x537d5581
@@ -519307,6 +519334,7 @@ interface {
   symbol_id: 0xf48c66e5
   symbol_id: 0x8b94484c
   symbol_id: 0x978bc797
+  symbol_id: 0xad5c16f2
   symbol_id: 0xfbb545c8
   symbol_id: 0xb5ce3771
   symbol_id: 0xffae3bcf
diff --git a/gki/aarch64/symbols/oplus b/gki/aarch64/symbols/oplus
index 5b994ea3b2bb..e9c63c9f2bd0 100644
--- a/gki/aarch64/symbols/oplus
+++ b/gki/aarch64/symbols/oplus
@@ -294,6 +294,7 @@
   __traceiter_android_vh_cpufreq_fast_switch
   __traceiter_android_vh_exit_check
   __traceiter_android_vh_move_entity
+  __traceiter_android_vh_scx_restore_flags
   __traceiter_android_vh_task_should_scx
   __traceiter_android_vh_scx_ops_consider_migration
   __traceiter_android_vh_scx_fix_prev_slice
@@ -436,6 +437,7 @@
   __tracepoint_android_vh_cpufreq_fast_switch
   __tracepoint_android_vh_exit_check
   __tracepoint_android_vh_move_entity
+  __tracepoint_android_vh_scx_restore_flags
   __tracepoint_android_vh_task_should_scx
   __tracepoint_android_vh_scx_ops_consider_migration
   __tracepoint_android_vh_scx_fix_prev_slice
-- 
2.50.1

